test/app/services/users.spec.js

```diff
@@ -0,0 +1,36 @@
+const { userMockReq, userMockRes } = require('../../mocks/users');
+const userService = require('../../../app/services/users');
+const errors = require('../../../app/errors');
+const UserRepository = require('../../../app/repositories/users');
+
+describe('Users Service', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should sign up an user successfully', async () => {
+    UserRepository.prototype.existBy = jest.fn().mockResolvedValue(false);
+    UserRepository.prototype.save = jest.fn().mockResolvedValue(userMockRes);
+
+    jest.spyOn(userService, 'signUp');
+    const response = await userService.signUp(userMockReq);
```

Comment:
is this mock necessary? Because you are mocking the database call and the services return the database response

Reply:
Yep, is really necessary mocking `save` repository function because we are testing the service function `scope`, then, every internal function that the `service function` uses, have to be mocked.